src/ledgerSigner/index.js

```diff
@@ -51,13 +52,20 @@ class LedgerNanoS {
     const address = await this.getEthAddress(accountIndex)
     const web3 = this.getWeb3()
     const balance = await web3.eth.getBalance(address)
-    return balance
+    return new BN(balance)
   }
 
   deviceConnected = async () => {
     try {
-      const address = await this.getEthAddress(0)
-      return address
+      return {
+        connected: false,
```

Comment:
connected: true,

Reply:
{
        connected: true,
        accounts: [{
          address: await this.getEthAddress(0),
          balance: {
            ethereum: await this.getBalance(0)
          }
        }],
        network: networkIdMap[process.env.REACT_APP_NETWORK_NAME]
      }